chain/network/src/concurrency/arc_mutex.rs

```diff
@@ -0,0 +1,27 @@
+use arc_swap::ArcSwap;
+use std::sync::{Arc, Mutex};
+
+/// Mutex which only synchronizes on writes.
+/// Reads always succeed and return the latest written version.
+pub struct ArcMutex<T> {
+    value: ArcSwap<T>,
+    mutex: Mutex<()>,
+}
+
+impl<T: Clone> ArcMutex<T> {
+    pub fn new(v: T) -> Self {
+        Self { value: ArcSwap::new(Arc::new(v)), mutex: Mutex::new(()) }
+    }
+    // non-blocking
+    pub fn load(&self) -> Arc<T> {
+        self.value.load_full()
+    }
+    // blocking
+    pub fn update<R>(&self, f: impl FnOnce(&mut T) -> R) -> R {
```

Comment:
Question: can we make this single-writer rather than blocking? Basically, use `.try_lock` rather than `.lock` here, and consider this to we a programming error if the caller doesn't enforce they are the single writer?

Reply:
I intend to actually have multiple concurrent writers (peers), just with a low probability of collision.

Reply:
However, I think I can replace this thing with the EbrCell you wrote about on zulip today.